modules/azurerm/Compute-Gallery/compute_gallery.tf

```diff
@@ -0,0 +1,19 @@
+# -------------------------------------------------------------------------------------
+#
+# Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com). All Rights Reserved.
+#
+# This software is the property of WSO2 LLC. and its suppliers, if any.
+# Dissemination of any information or reproduction of any material contained
+# herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+# You may not alter or remove any copyright or other notice from copies of this content.
+#
+# --------------------------------------------------------------------------------------
+
+resource "azurerm_shared_image_gallery" "compute_gallery" {
+  name                = join("_", ["sig", var.project, var.environment, var.sig_application_name])
```

Comment:
Let's join with `-`. Also let's allow users to override the name `sig-${var.name}`

Reply:
Azure compute gallery name must contain only English letters, numbers, underscores, and periods. The name cannot begin or end with underscores or periods.